### DISCUSSION TITLE
Lesion Processing considerations

### DISCUSSION THREAD

**Discussion by dbrennan44** [ftdc-picsl/antsnetct#44] [Ideas]
Hi there,

I'm processing TBI subjects with significant lesions. We have lesion masks, and I believe there's a BIDS-passing suffix for lesions masks. Could an option for altering the brain masks be included?

After (or within) cross-sectional pipeline, lesion masks are subtracted from brain extraction masks. 

`[longitudinal_pipeline.py 375] unified_mask_sst = ants_helpers.combine_masks(sst_t1w_masks, working_dir, thresh = **len(sst_t1w_masks)**) # get intersection instead of union of ses masks, so that lesion "holes" are propogated`

Downstream analyses can proceed as normal, doing "cost function masking" within the brain mask respecting the lesion. 

Sorry for roundabout way of proposing this, I'm still a novice when it comes to github/code dev. 

Thank you!

**Comment by cookpa** [comment]
This is a great idea. I will see what I can do about this.

I guess we would want to have two masks - the brain mask, which includes lesions but removes skull, etc. And a metric mask, which is [brain_mask - lesion_mask]. Thus lesions wouldn't count for registration or segmentation, but their values would still exist in the desc-brain images.

I don't know how lesion masks would play with the deep_atropos tools though. It might need retraining to deal with them.

**Reply by dbrennan44** [reply]
Previously I let deep_atropos run on the full brain mask, and just "cut" the lesion mask out of the produced segmentation images prior to running kelly kapowski. We have a priori information that the tissue is not GM or WM, so I feel justified in cutting it out. Then run KK as normal.

**Reply by cookpa** [reply]
That makes sense. deep_atropos does its own brain masking internally. I've not played with customizing the brain mask in a long time, but my first experiments in doing this didn't go well - changing the brain mask from what the training data expected made the segmentation noticeably worse.

If the lesions aren't undermining the rest of the segmentation, then masking them out after segmentation but before running KK might be the way to go.

**Reply by dbrennan44** [reply]
I have not found that the lesions change the segmentation anywhere else in the image, but I've only tested a few subjects. 

For example:
![image](https://github.com/user-attachments/assets/5a643c4f-0792-4d6e-b3b8-7d56cce7842e)

All ROIs outside the lesion area are 1-1 in terms of cortical thickness, with the lesion "cut" out of the image vs that same cutout in-painted.

**Reply by cookpa** [reply]
That's good to know - so it seems we could remove the lesions after deep_atropos and before thickness. 

I guess we could compute a longitudinal lesion mask as the union of all the session lesion masks, in SST space. Then warp the SST segmentation posteriors (minus lesions) to each session where they would be used for segmentation (minus lesions, again)

**Reply by dbrennan44** [reply]
I defer to your expertise on this front. 

What's tricky is if lesions or other pathology develop over time. You'll necessarily have only the tissue that is normal across all time points. Perhaps this is more ideal, as then you may have more confidence in the longitudinal estimates.

I wonder if subtraction of lesions in SST priors is even necessary, as you'll be subtracting the lesions in the session space posteriors anyways. The "easiest" implementation is to let the whole pipeline run as normal up until session-level KK, and just subtract the lesion from all segmentation/posterior estimates then run KK.
